java-backend/src/main/java/org/kframework/backend/java/kil/Rule.java

```diff
@@ -220,6 +220,11 @@ public boolean isConcrete() {
         return att().contains(Attribute.CONCRETE_FUNCTION_KEY);
     }
 
+    public boolean isConcrete(ImmutableSet<String> concreteRules) {
```

Comment:
This is worth renaming to `isConcreteRule`,and keep it with no arguments. To distinguish it from `isConcrete` inherited from `JavaSymbolicObject` which is called in different context and should be kept separated.